nservicebus/outbox/index.md

```diff
@@ -79,8 +79,61 @@ In order to gradually convert an entire system from the DTC to the Outbox:
 WARNING: When verifying outbox functionality, it can sometimes be helpful to temporarily [stop the MSDTC Service](https://technet.microsoft.com/en-us/library/cc770732.aspx). This ensures that the Outbox is working as expected, and no other resources are enlisting in distributed transactions.
 
 
+## Message identity and idempotent processing
+
+If handlers are idempotent then outbox might not be required at all. When considering not using the outbox and implementing idempotent processing keep in mind that any generated message *must* apply the same message identifier if invoked more-then-once to have deduplication working on the infrastructure level or to solve this at the logical level by validating message data. [Read more on how to set a message identifier.](/nservicebus/messaging/message-identity.md)
+
+
+## Concurrency
+
+Outbox cannot guarantee that handlers or sagas are invoked exactly once. Outbox does not prevent concurrent attempts to process the same message but ensures that the outcome of the processing is persisted only once for resources that share the outbox transaction.
+
+The following scenarios might cause handlers to be invoked multiple times:
+
+- Sender sends multiple physical message with the same message identifier (submit order form)
+- Transport can send multiple messages when having a failover, they operate in a at-least-once delivery model
+- The outbox dispatch stage can be invoked more-than-once if the previous dispatch (partially) failed and is retried.
+
+In those scenarios, multiple physical copies of the same logical message exist in the queue. If an endpoint allows concurrent message processing or is scaled out it can happen that these multiple copies are processed concurrently.
```

Comment:
> If an endpoint allows concurrent message processing or is scaled out it can happen that these multiple copies are processed concurrently

try
> If an endpoint allows concurrent message processing, or it is scaled out, its possible that the multiple copies of the message can be processed concurrently.